### PR TITLE
[MNT] Add metrics parameter for deep clustering module (#1945)

### LINKED ISSUE #1945: [MNT] Add metrics parameter for deep clustering module
### Describe the issue

The deep clustering module is missing the metrics parameter, in AEResNEet it exists but unused

### Suggest a potential alternative/fix

it should be default to None, and then set to ["mse"] if its None internally, the same way its done in deep clustering but here its mse instead of accuracy

### Additional context

Its for verbose purpose, doesnt affect anything else

### PR DESCRIPTION
**Pull request by neuron283**
#### Reference Issues/PRs
Fixes #1945 

#### What does this implement/fix?
Adds metrics parameter for deep clustering.

#### Does your contribution introduce a new dependency? If yes, which one?
No.

#### Any other comments?
Let me know if any changes required. Thanks!

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

### CODE REVIEW

**Comment by neuron283** [comment]
@all-contributors please add @neuron283 for maintenance

File: aeon/clustering/deep_learning/_ae_fcn.py [hunk 1]
```diff
@@ -201,6 +205,11 @@ def build_model(self, input_shape, **kwargs):
         import numpy as np
         import tensorflow as tf
 
+        if self.metrics is None:
+            metrics = ["mean_squared_error"]
+        else:
+            metrics = self.metrics
```
Comment (hadifawaz1999) [inline aeon/clustering/deep_learning/_ae_fcn.py:211]: the code should check if the input is not a list and put it in a list, as keras only takes input list for metrics. if the user gives a list it should stay a list, if its something else the code should raise a value error

File: aeon/clustering/deep_learning/_ae_fcn.py [hunk 2]
```diff
@@ -201,6 +205,11 @@ def build_model(self, input_shape, **kwargs):
         import numpy as np
         import tensorflow as tf
 
+        if self.metrics is None:
+            metrics = ["mean_squared_error"]
```
Comment (hadifawaz1999) [inline aeon/clustering/deep_learning/_ae_fcn.py:209]: can you call the variable metrics_ or _metrics ? and make it private with self ?

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
Thanks for taking care of this, can you do the same changes for AEResNet ? the parameter exists but unused yet

**Comment by neuron283** [comment]
> Thanks for taking care of this, can you do the same changes for AEResNet ? the parameter exists but unused yet

Sure. 
In AEResNet, the default value of metrics is ``` ["accuracy"] ``` . Should I keep that as it is or change it to ```mean_squared_error``` ?

**Comment by hadifawaz1999** [comment]
> > Thanks for taking care of this, can you do the same changes for AEResNet ? the parameter exists but unused yet
> 
> Sure. In AEResNet, the default value of metrics is `["accuracy"]` . Should I keep that as it is or change it to `["mean_squared_error"]` ?

it should be all the same way it was done for AEFCN, the same way you did it in this PR for now

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
It seems the PR reverted the changes from the latest PR we got in #1944

**Comment by hadifawaz1999** [comment]
image
the issues seems here, for some reason when mergin the main branch it reverted the latest commit to main

**Comment by neuron283** [comment]
> image the issues seems here, for some reason when mergin the main branch it reverted the latest commit to main

Sorry for inconvenience, I'll open a new PR and make the necessary changes.

**Pull request closed without merging**